app/code/Magento/Customer/view/frontend/web/js/customer-data.js

```diff
@@ -408,5 +408,21 @@ define([
         }
     });
 
+    /**
+     * Check quote was updated after page is loaded, if yes, invalidate and reload cart cache
+     */
+    $(document).ready(function () {
```

Comment:
Seems like it will add +1 not cacheable request to backend for all pages that will significantly increase load on servers. I don't think it's really good option

Reply:
I thought in many solutions and the only solution I found was that, there isn't no way to solve this issue without having at least a fresh quote.updated_at comparing with localStorage from browser.

The endpoint /customer/section/load is called every time we add products to cart (submit) and during ajaxComplete event.

Reply:
I put this for discussion with other maintainers

Reply:
From @sidolov in Slack:
> Hi Ihor, to be honest, we have a story in our internal backlog to implement the cart invalidation mechanism when some information was changed on the backend. It’s not implemented yet and we don’t have any estimates when it will be implemented. **Additional request on all pages definitely not acceptable since will affect overall storefront performance**. I believe the right way is to investigate the real impact of this issue on merchants, maybe it’s something we can live with or in case with real annoying problem we should try to prioritize the story with product organization, introduce the design for feature and start implementation.

Reply:
@ihor-sviziev + @sidolov 

The scenario exposed on issue will happen depending on how many times the merchant or integration updates the product price per day or hour or N minutes, it'll be one additional request if the customer has the product that was updated the price on cart and the localStorage isn't expired yet, so it doesn't happen every time (additional request), that is why I don't see the solution won't affect overall storefront performance.

Observation: the invalidation and reload of cart cache (localStorage) happens every time that customer adds some product into the cart, so why doesn't have any concern about overall storefront performance for this scenario, if it's the same made by solution on a specific scenario (issue scenario)?

1) the scenario didn't happen, 10 products were added to cart = 10 requests <<< there is "no" concern about overall storefront performance

2) the scenario happened, 1 product on cart = 2 request <<< "there is" concern about overall storefront performance

Reply:
Just an idea - what if we'll add some thin controller that will return kind of "updated at" date (one value for all products) and cache this result on Full page cache, this value will be updated automatically when the any product changes.
What do you thing?

Reply:
One more idea - add some esi block to the all pages that will contain small piece of js, like `var PRODUCTS_UPDATED_AT = "2020-10-23 06:46";`. As result this block will be included to all pages and will not affect not the page load time not bring additional requests to backend.